isce2_topsapp/__main__.py

```diff
@@ -37,33 +39,54 @@ def localize_data(reference_scenes: list,
     return out
 
 
+def ensure_earthdata_credentials(username: Optional[str] = None, password: Optional[str] = None,
+                                 host: str = 'urs.earthdata.nasa.gov', netrc_file: Path = Path.home() / '.netrc'):
+    """Ensures an Earthdata username and password are provided
+
+     Earthdata username and password have been provided by, in order of preference, one of:
+        * `netrc_file`
+        * `username` and `password`
+        * `EARTHDATA_USERNAME` and `EARTHDATA_PASSWORD` environment variables
+    """
+    if username is None:
+        username = os.getenv('EARTHDATA_USERNAME', '')
+
+    if password is None:
+        password = os.getenv('EARTHDATA_PASSWORD', '')
+
+    if not netrc_file.exists():
+        netrc_file.write_text(f'machine {host} login {username} password {password}')
+        netrc_file.chmod(0o000600)
+
+    try:
+        dot_netrc = netrc.netrc(netrc_file)
+        username, _, password = dot_netrc.authenticators(host)
+    except (netrc.NetrcParseError, TypeError):
+        raise ValueError(
+            f'Please provide valid Earthdata login credentials via {netrc_file}, '
+            f'the --username and --password CLI option, or '
+            f'the EARTHDATA_USERNAME and EARTHDATA_PASSWORD environment variables.'
+        )
+
+    return username, password
+
+
```

Comment:
Beautiful! So, all the readme instructions about netrc are valid and so are the command line scripts that use --username/--password. Thanks Joe.

Reply:
yeah! Should just extend functionality